Packet++/header/HttpLayer.h

```diff
@@ -207,201 +207,230 @@ namespace pcpp
+			Http450BlockedByWindowsParentalControls = 450,
 			/** 451 (various messages) */
-			Http451,
+			Http451 = 451,
 			/** 494 Request Header Too Large */
-			Http494RequestHeaderTooLarge,
+			Http494RequestHeaderTooLarge = 494,
 			/** 495 Cert Error */
-			Http495CertError,
+			Http495CertError = 495,
 			/** 496 No Cert */
-			Http496NoCert,
+			Http496NoCert = 496,
 			/** 497 HTTP to HTTPS */
-			Http497HTTPtoHTTPS,
+			Http497HTTPtoHTTPS = 497,
 			/** 498 Token expired/invalid */
-			Http498TokenExpiredInvalid,
+			Http498TokenExpiredInvalid = 498,
 			/** 499 (various messages) */
-			Http499,
+			Http499 = 499,
 			/** 500 Internal Server Error */
-			Http500InternalServerError,
+			Http500InternalServerError = 500,
 			/** 501 Not Implemented */
-			Http501NotImplemented,
+			Http501NotImplemented = 501,
 			/** 502 Bad Gateway */
-			Http502BadGateway,
+			Http502BadGateway = 502,
 			/** 503 Service Unavailable */
-			Http503ServiceUnavailable,
+			Http503ServiceUnavailable = 503,
 			/** 504 Gateway Timeout */
-			Http504GatewayTimeout,
+			Http504GatewayTimeout = 504,
 			/** 505 HTTP Version Not Supported */
-			Http505HTTPVersionNotSupported,
+			Http505HTTPVersionNotSupported = 505,
 			/** 506 Variant Also Negotiates */
-			Http506VariantAlsoNegotiates,
+			Http506VariantAlsoNegotiates = 506,
 			/** 507 Insufficient Storage */
-			Http507InsufficientStorage,
+			Http507InsufficientStorage = 507,
 			/** 508 Loop Detected */
-			Http508LoopDetected,
+			Http508LoopDetected = 508,
 			/** 509 Bandwidth Limit Exceeded */
-			Http509BandwidthLimitExceeded,
+			Http509BandwidthLimitExceeded = 509,
 			/** 510 Not Extended */
-			Http510NotExtended,
+			Http510NotExtended = 510,
 			/** 511 Network Authentication Required */
-			Http511NetworkAuthenticationRequired,
+			Http511NetworkAuthenticationRequired = 511,
 			/** 520 Origin Error */
-			Http520OriginError,
+			Http520OriginError = 520,
 			/** 521 Web server is down */
-			Http521WebServerIsDown,
+			Http521WebServerIsDown = 521,
 			/** 522 Connection timed out */
-			Http522ConnectionTimedOut,
+			Http522ConnectionTimedOut = 522,
 			/** 523 Proxy Declined Request */
-			Http523ProxyDeclinedRequest,
+			Http523ProxyDeclinedRequest = 523,
 			/** 524 A timeout occurred */
-			Http524aTimeoutOccurred,
+			Http524aTimeoutOccurred = 524,
 			/** 598 Network read timeout error */
-			Http598NetworkReadTimeoutError,
+			Http598NetworkReadTimeoutError = 598,
 			/** 599 Network connect timeout error */
-			Http599NetworkConnectTimeoutError,
+			Http599NetworkConnectTimeoutError = 599,
 			/** Unknown status code */
-			HttpStatusCodeUnknown
+			HttpStatusCodeUnknown = 999999
+		};
+
+		HttpResponseStatusCode() = default;
+		constexpr HttpResponseStatusCode(Value statusCode) : m_value(statusCode) { }
+		
+		constexpr operator Value() const { return m_value; }
+		explicit operator bool() const = delete;
+
+		struct HttpResponseStatusCodeHash {
```

Comment:
for `unordered_map`

Reply:
> > Why is this needed for the unordered_map? Can't it hash by enum value?
> 
> unfortunately, it's needed, because we need to tell unordered_map how to hash `HttpResponseStatusCode` (even though `HttpResponseStatusCode` can implicitly cast to int)

In that case, let's move the hash function to `HttpLayer.cpp`? I don't think it should be part of `HttpResponseStatusCode `